database/database.js

```diff
@@ -74,17 +115,55 @@ const createBasePrice = () => {
   range = range / 10;
 
   // Construct basePrice
-  let basePrice = (Math.floor(Math.random() * range) * 10 + minPrice);
-
-  // Get rid of anything after 2 decimal places (cleaning up after floating-point issues)
-  basePrice = Math.round(basePrice * 100) / 100;
+  let basePrice = (Math.floor(Math.random() * range) * 10 + Math.floor(minPrice) + 0.99);
 
   // 20-ish% of the time we'll add $5, so we get the occasional $54.99 or whatever
-  if (Math.random() < 0.2) {
+  if (randomDecider(20)) {
     basePrice = basePrice + 5;
   }
 
+  // Get rid of anything after 2 decimal places (cleaning up after floating-point issues)
+  basePrice = Math.round(basePrice * 100) / 100;
+
   return basePrice;
 }
 
-exports.populateDatabase = populateDatabase;
\ No newline at end of file
+const generatePreviewVideoData = (index) => {
+
+  const videoIndex = Math.floor(Math.random() * 10);
+
+  const previewVideoData = new PreviewVideo({
+    courseID: index,
+    previewVideoUrl: "https://example.com/previewVideo" + videoIndex + ".mp4",
+  });
+
+  return previewVideoData;
+}
+
+const generateSidebarData = (index) => {
+
+  const sidebarData = new Sidebar({
+    courseID: index,
+    fullLifetimeAccess: randomDecider(70) ? "Full lifetime access" : "Full access during subscription term",
+    accessTypes: "Access on mobile and TV",
+    assignments: randomDecider(70),
+    certificateOfCompletion: randomDecider(90)
+  });
+
+  return sidebarData;
+}
+
+const randomDecider = (percentageChance) => {
```

Comment:
Not really important, but eslint/pomander prefer functions to be defined above where they're called, although JS doesn't really seem to care